packages/components/src/ui/ariakit-tooltip/index.tsx

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import { Tooltip, TooltipAnchor, useTooltipState } from 'ariakit/tooltip';
+
+/**
+ * WordPress dependencies
+ */
+import { cloneElement, isValidElement } from '@wordpress/element';
+import { useInstanceId } from '@wordpress/compose';
+import deprecated from '@wordpress/deprecated';
+
+/**
+ * Internal dependencies
+ */
+import { TOOLTIP_DELAY } from '../../tooltip/';
+import type { ToolTipProps } from './types';
+import Shortcut from '../../shortcut';
+import { positionToPlacement as __experimentalPopoverLegacyPositionToPlacement } from '../../popover/utils';
+import * as styles from './styles';
+import { contextConnectWithoutRef } from '../context/context-connect';
+import { useCx } from '../../utils/hooks/use-cx';
+
+function AriaToolTip( props: ToolTipProps ) {
+	const {
+		children,
+		delay = TOOLTIP_DELAY,
+		placement,
+		position,
+		shortcut,
+		text,
+	} = props;
+
+	const baseId = useInstanceId( ToolTip, 'tooltip' );
+	const describedById = text || shortcut ? baseId : undefined;
+
+	const DEFAULT_PLACEMENT = 'bottom';
+
+	// Compute tooltip's placement:
+	// - give priority to `placement` prop, if defined
+	// - otherwise, compute it from the legacy `position` prop (if defined)
+	// - finally, fallback to the DEFAULT_PLACEMENT.
+	let computedPlacement;
+	if ( placement !== undefined ) {
+		computedPlacement = placement;
+	} else if ( position !== undefined ) {
+		computedPlacement =
+			// @ts-expect-error
+			__experimentalPopoverLegacyPositionToPlacement( position );
+	}
+	computedPlacement = computedPlacement || DEFAULT_PLACEMENT;
+
+	if ( position !== undefined ) {
+		deprecated( '`position` prop in wp.components.tooltip', {
+			since: '6.3',
+			alternative: '`placement` prop',
+		} );
+	}
+
+	const tooltipState = useTooltipState( {
+		placement: computedPlacement,
+		timeout: delay,
+	} );
+
+	const cx = useCx();
+	const ToolTipClassName = cx( styles.ToolTip );
+	const ShortcutClassName = cx( styles.Shortcut );
+
+	return (
+		<>
+			<TooltipAnchor described state={ tooltipState }>
```

Comment:
After seeing [this comment](https://github.com/WordPress/gutenberg/issues/48222#issuecomment-1482937115), I just wanted to add a note to clarify what the `described` prop is. In ariakit, `aria-labelledby` is the default but `aria-describedby` can be added by adding the `described` prop: 
https://github.com/ariakit/ariakit/blob/5d8a1f047fcadcf117073c70359663a3946b73bf/packages/ariakit/src/tooltip/tooltip-anchor.ts#L122-L136

According to ARIA guidelines, `aria-describedby` is required for tooltip so I'm not sure why it isn't the default option. @ciampo pointed out it might be worth asking the author about this choice.